src/mqt/yaqs/core/data_structures/networks.py

```
@@ -1775,6 +1776,61 @@ def to_matrix(self) -> NDArray[np.complex128]:
+        over the terms in the MPO sum. This avoids creating the full dense matrix
+        intermediate.
+
+        Returns:
+            The sparse matrix representation of the MPO in CSR format.
+        """
+        d = self.physical_dimension
+
+        current_operators = {0: scipy.sparse.csr_matrix(np.eye(1, dtype=complex))}
+
+        for tensor in self.tensors:
+            _d_out, _d_in, D_left, D_right = tensor.shape  # noqa: N806
```

Ruff (0.15.1)

[warning] Unused `noqa` directive (non-enabled: `N806`)

Remove unused `noqa` directive

(RUF100)